common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx

```diff
@@ -0,0 +1,153 @@
+
+export default class ConfirmationModal extends React.Component {
+  props: Props;
+  state: State;
+
+  constructor(props: Props) {
+    super(props);
+    const { value, gasPrice, nonce } = props.rawTransaction;
+
+    // TODO: Fix with wallet address promise
+    this.state = {
+      address: props.wallet.getAddress(),
+      value: toUnit(new Big(value, 16), 'wei', 'ether').toString(),
+      gasPrice: toUnit(new Big(gasPrice, 16), 'wei', 'gwei').toString(),
+      nonce: new Big(nonce, 16).toString(),
+      timeToRead: 5
+    };
+  }
+
+  // Count down 5 seconds before allowing them to confirm
+  readTimer = null;
+  componentDidMount() {
+    this.readTimer = setInterval(() => {
+      if (this.state.timeToRead > 0) {
+        this.setState({ timeToRead: this.state.timeToRead - 1 });
+      } else {
+        clearTimeout(this.readTimer);
+      }
+    }, 1000);
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.readTimer);
+  }
+
+  _confirm() {
+    if (this.state.timeToRead < 1) {
+      this.props.onConfirm(this.props.rawTransaction);
+    }
+  }
+
+  render() {
+    const { node, rawTransaction, onCancel } = this.props;
+    const { address, value, gasPrice, timeToRead } = this.state;
+
+    const buttonPrefix = timeToRead > 0 ? `(${timeToRead}) ` : '';
+    const buttons = [
+      {
+        text: buttonPrefix + translate('SENDModal_Yes'),
+        type: 'primary',
+        disabled: timeToRead > 0,
+        onClick: this._confirm()
+      },
+      {
+        text: translate('SENDModal_No'),
+        type: 'default',
+        onClick: onCancel
+      }
+    ];
+
+    return (
+      <Modal
+        title="Confirm Your Transaction"
+        buttons={buttons}
+        handleClose={onCancel}
+        isOpen={true}
+      >
+        <div className="ConfModal">
+          <div className="ConfModal-summary">
+            <div className="ConfModal-summary-icon ConfModal-summary-icon--from">
+              <Identicon size="100%" address={address} />
+            </div>
+            <div className="ConfModal-summary-amount">
+              <div className="ConfModal-summary-amount-arrow" />
+              <div className="ConfModal-summary-amount-currency">
+                {value} ETH
+              </div>
+            </div>
+            <div className="ConfModal-summary-icon ConfModal-summary-icon--to">
+              <Identicon size="100%" address={rawTransaction.to} />
+            </div>
+          </div>
+
+          <ul className="ConfModal-details">
+            <li className="ConfModal-details-detail">
+              You are sending to <code>{rawTransaction.to}</code>
+            </li>
+            <li className="ConfModal-details-detail">
+              You are sending <strong>{value} ETH</strong> with a gas price of{' '}
+              <strong>{gasPrice} gwei</strong>
+            </li>
+            <li className="ConfModal-details-detail">
+              You are transmitting on the <strong>{node.network}</strong>{' '}
```

Comment:
Not sure where I came up with this copy, original is `interacting with the` which sounds way less nerdy. I'll change it back.